### PR TITLE
Implement asfortranarray

### PR DESCRIPTION
**Pull request by okuta**

### CODE REVIEW

File: tests/cupy_tests/manipulation_tests/test_kind.py [hunk 1]
```diff
@@ -7,3 +7,15 @@
 class TestKind(unittest.TestCase):
 
     _multiprocess_can_split_ = True
+
+    @testing.for_all_dtypes()
+    @testing.numpy_cupy_array_equal(type_check=False)
+    def test_asfortranarray1(self, xp, dtype):
+        x = xp.zeros(2, 3)
```
Comment (delta2323) [inline tests/cupy_tests/manipulation_tests/test_kind.py]: It should be `xp.zeros((2, 3))`. So this caused an error, but it is caught and grasped [here](https://github.com/pfnet/chainer/blob/master/cupy/testing/helper.py#L24).

File: cupy/manipulation/kind.py [hunk 2]
```diff
@@ -1,11 +1,26 @@
...
+        dtype (str or dtype object, optional): By default, the data-type is
+            inferred from the input data.
+
+    Returns:
+        ~cupy.ndarray: The input `a` in Fortran, or column-major, order.
+
+    .. seealso:: :func:`numpy.asfortranarray`
+
+    """
+    ret = cupy.empty(a.shape.T, a.dtype if dtype is None else dtype)
```
Comment (delta2323) [inline cupy/manipulation/kind.py]: `tuple` does not have an attribute `T`.

Comment (delta2323) [inline cupy/manipulation/kind.py]: What happens if we feed `a` with a fortran array?

File: tests/cupy_tests/manipulation_tests/test_kind.py (hunk 1 above)
Comment (okuta) [inline tests/cupy_tests/manipulation_tests/test_kind.py]: Thank you! 
I will fix.

**Comment by okuta** [comment]
@delta2323 I fixed.

File: tests/cupy_tests/manipulation_tests/test_kind.py [hunk 3]
```diff
@@ -7,3 +7,21 @@
 class TestKind(unittest.TestCase):
 
     _multiprocess_can_split_ = True
+
+    @testing.for_all_dtypes()
+    @testing.numpy_cupy_array_equal(type_check=False)
```
Comment (delta2323) [inline tests/cupy_tests/manipulation_tests/test_kind.py]: Why `type_check` is disabled?

**Comment by delta2323** [comment]
Please add tests in which `ndarray.flags` are set correctly after the operation.

**Comment by delta2323** [comment]
@okuta I updated comments.

File: tests/cupy_tests/manipulation_tests/test_kind.py (hunk 3 above)
Comment (okuta) [inline tests/cupy_tests/manipulation_tests/test_kind.py]: This test checks the stride. It is a tuple.

**Comment by delta2323** [comment]
LGTM :+1:

**Pull request merged**